src/compiler/checker.ts

```diff
@@ -5962,21 +5962,26 @@ namespace ts {
                 return !(getObjectFlags(type) & ObjectFlags.Reference) || !isTypeReferenceNode(existing) || length(existing.typeArguments) >= getMinTypeArgumentCount((type as TypeReference).target.typeParameters);
             }
 
+            function tryGetTypeFromDeclarationWithTypeAnnotation(symbol: Symbol, enclosingDeclaration: Node | undefined) {
+                const declaration = getDeclarationWithTypeAnnotation(symbol, enclosingDeclaration);
+                if (declaration) {
+                    return isFunctionLikeDeclaration(declaration) && !isGetAccessorDeclaration(declaration) ? undefined :
+                        getEffectiveTypeAnnotationNode(declaration);
+                }
+            }
+
             /**
              * Unlike `typeToTypeNodeHelper`, this handles setting up the `AllowUniqueESSymbolType` flag
              * so a `unique symbol` is returned when appropriate for the input symbol, rather than `typeof sym`
              */
             function serializeTypeForDeclaration(context: NodeBuilderContext, type: Type, symbol: Symbol, enclosingDeclaration: Node | undefined, includePrivateSymbol?: (s: Symbol) => void, bundled?: boolean) {
                 if (type !== errorType && enclosingDeclaration) {
-                    const declWithExistingAnnotation = getDeclarationWithTypeAnnotation(symbol, enclosingDeclaration);
-                    if (declWithExistingAnnotation && !isFunctionLikeDeclaration(declWithExistingAnnotation)) {

```

Comment:
As the change currently stands I'm not a huge fan of extracting this function. It makes the diff pretty hard to read.
On the other hand, if adding `set` tests shows that the code needs to be more complex, it might tip the balance more strongly toward having a separate function.

Reply:
Ok, removed function